docs_src/design/adr/device-service/0011-DeviceService-Rest-API.md

```diff
@@ -144,9 +154,13 @@ ie, `new-value = (current-value & !mask) | request-value`
 
 The combination of mask and shift can therefore be used to access data contained in a subdivision of an octet.
 
+It is possible that following the application of the specified transformations, a value may exceed the range that may be represented by its type. Should this occur on a set operation, a suitable error should be logged and returned, along with the `Bad Request` http code 400. If it occurs as part of a get operation, the Reading's value should be set to the String `"overflow"`.
+
 #### Assertions and Mappings 
 
-Assertions are another attribute in a device resource's PropertyValue which specify a string value which the result is compared against. If the comparison fails, then the result is set to a string of the form *"Assertion failed for device resource: \<name>, with value: \<result>"*, this also has a side-effect of setting the device operatingstate to `DISABLED`. A 500 status code is also returned.
+Assertions are another attribute in a device resource's PropertyValue which specify a string value which the result is compared against. If the comparison fails, then the result is set to a string of the form *"Assertion failed for device resource: \<name>, with value: \<result>"*, this also has a side-effect of setting the device operatingstate to `DISABLED`. A 500 status code is also returned. Note that the error response and status code should be returned regardless of the `ds-returnevent` setting.
```

Comment:
The use of result is a bit confusing here. It's really the `Reading.Value` (after all transforms have been applied) that's being compared to the assertion.

What happens if multiple readings are being returned? Does the failure of a single assertion trigger 500 status code, and its up to the receiver to walk through the readings as some may be valid?